evm/p2p/kademlia.py

```diff
@@ -191,10 +191,13 @@ def __contains__(self, node):
     def __len__(self):
         return len(self.nodes)
 
-    def __le__(self, other):
+    def __lt__(self, other):
         if not isinstance(other, self.__class__):
             raise TypeError("Cannot compare KBucket with type {}.".format(other.__class__))
-        return self.end <= other.end
+        # Check for invalid state of KBuckets
+        if not self.end < other.start:
+            raise ValueError("Invalid Buckets.")
+        return self.end < other.start
```

Comment:
If I'm reading this code right, this method can only return True or a ValueError, but never False. Is that what you intended, and, if so, why?

Reply:
Nope, good catch. Was just tinkering with a way to check for invalid state of KBuckets - forgot it was there when I pushed.